paasta_tools/tron_tools.py

```diff
@@ -0,0 +1,371 @@
+                    'in service {service}'.format(
+                        action=action_dict.get('name'),
+                        job=self.get_name(),
+                        deploy_group=action_deploy_group,
+                        service=action_service,
+                    ),
+                )
+        else:
+            branch_dict = {}
+
+        if 'cluster' not in action_dict:
+            action_dict['cluster'] = default_paasta_cluster
+
+        return TronActionConfig(
+            service=action_service,
+            instance=compose_instance(self.get_name(), action_dict.get('name')),
+            config_dict=action_dict,
+            branch_dict=branch_dict,
+            soa_dir=self.soa_dir,
+        )
+
+    def get_actions(self, default_paasta_cluster):
+        if self._actions:
+            return self._actions
+
+        actions = []
+        for action_dict in self.config_dict.get('actions'):
+            actions.append(self._get_action_config(action_dict, default_paasta_cluster))
+        self._actions = actions
+        return self._actions
+
+    def get_cleanup_action(self, default_paasta_cluster):
+        action_dict = self.config_dict.get('cleanup_action')
+        if not action_dict:
+            return None
+        return self._get_action_config(action_dict, default_paasta_cluster)
+
+    def format_tron_job_dict(self, cluster_fqdn_format, default_paasta_cluster):
+        action_dicts = [
+            action_config.format_tron_action_dict(cluster_fqdn_format)
+            for action_config in self.get_actions(default_paasta_cluster)
+        ]
+
+        job_config = {
+            'name': self.get_name(),
+            'node': self.get_node(),
+            'schedule': self.get_schedule(),
+            'actions': action_dicts,
+            'monitoring': self.get_monitoring(),
+            'queueing': self.get_queueing(),
+            'run_limit': self.get_run_limit(),
+            'all_nodes': self.get_all_nodes(),
+            'enabled': self.get_enabled(),
+            'allow_overlap': self.get_allow_overlap(),
+            'max_runtime': self.get_max_runtime(),
+            'time_zone': self.get_time_zone(),
+        }
+        cleanup_config = self.get_cleanup_action(default_paasta_cluster)
+        if cleanup_config:
+            job_config['cleanup_action'] = cleanup_config.format_tron_action_dict(cluster_fqdn_format)
+
+        # Only pass non-None values, so Tron will use defaults for others
+        return {key: val for key, val in job_config.items() if val is not None}
+
+    def __eq__(self, other):
+        if isinstance(other, type(self)):
+            return self.config_dict == other.config_dict
+        return False
+
+
+def load_tron_service_config(service, tron_cluster, soa_dir=DEFAULT_SOA_DIR):
+    """Load all configured jobs for a service, and any additional config values."""
+    tron_conf_file = os.path.join(
+        os.path.abspath(soa_dir), 'tron', tron_cluster, service + '.yaml',
```

Comment:
hm, I thought the tron-{cluster}.yaml file would be in each service directory?

If I'm reading this correctly, it looks in /nail/etc/services/tron/prod/myservice.yaml - but I was expecting us to do /nail/etc/services/myservice/tron-prod.yaml ?

Reply:
You're reading it correctly, it's looking for /nail/etc/services/tron/prod/myservice.yaml right now. It should be changed to the 2nd thing after we migrate the tronfig files to service directories, but I want to be able to use this script before we finish that migration.